test/config.js

```diff
@@ -0,0 +1,76 @@
+import path from 'path';
+import test from 'ava';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+
+const fixtureBasePath = path.resolve('test', 'fixtures', 'config');
+
+const callConfigWhenGlobalBinaryUsed = async homedirStub => {
+	const pathsPkgDir = [path.resolve(fixtureBasePath, 'pkg-dir'),
+		path.resolve(fixtureBasePath, 'local1'),
+		path.resolve(fixtureBasePath, 'local2')];
+
+	const promises = [];
+	pathsPkgDir.forEach(pathPkgDir => {
+		promises.push(proxyquire('../source/config', {
+			'is-installed-globally': true,
+			'pkg-dir': async () => {
+				return pathPkgDir;
+			},
+			os: {
+				homedir: homedirStub
+			}
+		})());
+	});
+	return Promise.all(promises);
+};
+
+const callConfigWhenLocalBinaryUsed = async pathPkgDir => {
+	const homedirs = [path.resolve(fixtureBasePath, 'homedir1'),
+		path.resolve(fixtureBasePath, 'homedir2')];
+
+	const promises = [];
+	homedirs.forEach(homedir => {
+		promises.push(proxyquire('../source/config', {
+			'is-installed-globally': false,
+			'pkg-dir': async () => {
+				return pathPkgDir;
+			},
+			os: {
+				homedir: () => {
+					return homedir;
+				}
+			}
+		})());
+	});
+	return Promise.all(promises);
+};
+
+test('should always return config from home-directory when global binary used and `.np-config-json` in home-directory exists', async t => {
```

Comment:
I'd rephrase this to

```suggestion
test('returns config from home-directory when global binary used and `.np-config-json` in home-directory exists', async t => {
```

Reply:
(Same for the rest of the test names.)

Reply:
I ended up fixing this myself in e68c601d35c85b24b2af729f9a4e9a9f38cfb38b to speed things up :)